src/main/java/seedu/address/logic/parser/AddTransactionCommandParser.java

```diff
@@ -23,43 +26,56 @@
  */
 public class AddTransactionCommandParser implements Parser<AddTransactionCommand> {
 
+    public static final String VALIDATION_REGEX = String.format("^%s%s %s%s %s%s (%s%s %s%s)+$",
+            PREFIX_DESCRIPTION, trimRegExp(Description.VALIDATION_REGEX),
+            PREFIX_NAME, trimRegExp(Name.VALIDATION_REGEX), PREFIX_COST, trimRegExp(Amount.VALIDATION_REGEX),
+            PREFIX_NAME, trimRegExp(Name.VALIDATION_REGEX), PREFIX_WEIGHT, trimRegExp(Weight.VALIDATION_REGEX));
+
+    public static final String MESSAGE_POSITIVE_VALUES_RESTRICTION = "All values must be positive";
+
     /**
      * Parses the given {@code String} of arguments in the context of the AddTransactionCommand
      * and returns an AddTransactionCommand object for execution.
      * @throws ParseException if the user input does not conform the expected format.
      */
     public AddTransactionCommand parse(String args) throws ParseException {
         ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_COST, PREFIX_DESCRIPTION);
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_COST, PREFIX_DESCRIPTION, PREFIX_WEIGHT);
 
         if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_COST, PREFIX_DESCRIPTION)
                 || !argMultimap.getPreamble().isEmpty()) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
                     AddTransactionCommand.MESSAGE_USAGE));
         }
 
-        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_NAME, PREFIX_COST, PREFIX_DESCRIPTION);
-        Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
-        String costString = argMultimap.getValue(PREFIX_COST).get();
-        //Checking if amount is negative and then stripping the negative sign. This is to swap the payee and person that
-        //owes money when creating the transaction.
-        boolean isNegative = false;
-        if (costString.charAt(0) == '-') {
-            isNegative = true;
-            costString = costString.substring(1);
+        if (!args.trim().matches(VALIDATION_REGEX)) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    AddTransactionCommand.MESSAGE_USAGE));
+        }
+        List<String> names = argMultimap.getAllValues(PREFIX_NAME);
+        List<String> weights = argMultimap.getAllValues(PREFIX_WEIGHT);
+        assert names.size() == weights.size() + 1;
```

Comment:
Can consider throwing an exception here (with a message) instead since it is an external error caused by user

Reply:
By right the regex should ensure that this is true